cmd/rpcdaemon/commands/daemon.go

```diff
@@ -18,10 +18,11 @@ func APIList(db kv.RoDB, borDb kv.RoDB, eth rpchelper.ApiBackend, txPool txpool.
 	filters *rpchelper.Filters, stateCache kvcache.Cache,
 	blockReader services.FullBlockReader, agg *libstate.AggregatorV3, cfg httpcfg.HttpCfg, engine consensus.EngineReader,
 	l2RpcUrl string,
+	allowFreeTransactions, allowPreEIP155Transactions bool,
 ) (list []rpc.API) {
 	base := NewBaseApi(filters, stateCache, blockReader, agg, cfg.WithDatadir, cfg.EvmCallTimeout, engine, cfg.Dirs)
 	base.SetL2RpcUrl(l2RpcUrl)
-	ethImpl := NewEthAPI(base, db, eth, txPool, mining, cfg.Gascap, cfg.ReturnDataLimit, "")
+	ethImpl := NewEthAPI(base, db, eth, txPool, mining, cfg.Gascap, cfg.ReturnDataLimit, "", allowFreeTransactions, allowPreEIP155Transactions)
```

Comment:
there are a few parameters that are zk-related, maybe group then into a struct? l1rpc, and these flags?

Reply:
All zk-related are currently in Zk struct. There are 22 flags, some of them are prefixed by "L2" or "L1" and some are non-prefixed. By saying a "struct" do you really mean to make a sub-struct inside Zk struct or you meant adding a prefix?

Reply:
There were changes merged yesterday which could help here.  We already pass the l2RPCUrl and now with two more ZK related things it makes sense to just pass in the full `ZK` config struct and to add these new properties to that.  If you rebase this branch onto zkevm you'll see the changes made there as they'll conflict

Reply:
I will update it so it will use the entire struct

Reply:
Done